happi/cli.py

```diff
@@ -102,13 +105,58 @@ def happi_cli(args):
                 return
             if value.replace('.', '').isnumeric():
                 logger.debug('Changed %s to float', value)
-                value = float(value)
-            client_args[criteria] = value
+                value = str(float(value))
 
-        results = client.search(**client_args)
-        if results:
+            if criteria == 'z' and ',' in value:
+                start = None
+                end = None
+                z_is_range = True
+                try:
+                    start, end = value.replace(
+                        '[', '').replace(']', '').split(',')
+                    start = float(start)
+                    end = float(end)
+                except Exception as ex:
+                    logger.error("Invalid numbers for the z range %s", ex)
+                    sys.exit(1)
+                if start < end:
+                    results += client.search_range('z', start, end)
+                else:
+                    logger.error('Invalid z range')
+
+            # skip the criteria for range values
+            # it won't be a valid criteria for search_regex()
+            if z_is_range and criteria == 'z':
+                pass
+            else:
+                client_args[criteria] = fnmatch.translate(value)
+
+        results += client.search_regex(**client_args)
+
+        # find the repeated items
+        res_size = len(results)
+        repeated = []
+        for i in range(res_size):
+            k = i + 1
+            for j in range(k, res_size):
+                if results[i] == results[j] and results[i] not in repeated:
+                    repeated.append(results[i])
+
+        # we only want to return the ones that have been repeated
+        # they have been matched with both serch_regex() & search_range()
+        if repeated:
+            for res in repeated:
+                res.item.show_info()
+            return repeated
+        # only matched with serach_regex()
```

Comment:
you've got some typos in your comment here and a few lines above

Reply:
thanks for catching that